### ISSUE TITLE
Deprecated error in Opencats

### ISSUE THREAD

**Issue by karthy** [opencats/OpenCATS#5]
When i installed Opencats 0.9.1a in windows 7 and run it on Chrome then it's shown

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\web_tester.php on line 506

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\test_case.php on line 74

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\test_case.php on line 76

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\test_case.php on line 444

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\test_case.php on line 557

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\VTG\Projects\OB_CAT\lib\simpletest\test_case.php on line 604

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\expectation.php on line 33

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\browser.php on line 180

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\browser.php on line 192

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\http.php on line 101

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\VTG\Projects\OB_CAT\lib\simpletest\http.php on line 103

Deprecated: Assigning the return value of new by reference is deprecated in C:\xampp\htdocs\..........\lib\simpletest\http.php on line 282

and etc

**Comment by opencats** [comment]
Hi Karthy, this will be fixed when I roll in the fixes from crossgate9

**Comment by manilal** [comment]
@RussH I was able to reproduce these errors in my Fedora 21 box running PHP-5.6.4

**Comment by RussH** [comment]
Hi @libregeek was that using the cats-0.9.2 code after commit https://github.com/opencats/OpenCATS/commit/783360df5cc08104767189e1161efe7191b69310 ?

**Comment by libregeek** [comment]
In my working copy, git log says f08a6e34f9ae947c2fde5db17f9a78a2bd352509 is the last one. May be I should fetch the latest master and re-test it.

**Comment by libregeek** [comment]
I tried the latest copy and it throws the following exceptions during signin and logout:

```
[Wed Feb 11 13:34:59.884174 2015] [:error] [pid 1345] [client 127.0.0.1:55235] PHP Deprecated:  Assigning the return value of new by reference is deprecated in /var/www/html/demo/opencats/lib/simpletest/parser.php on line 590, referer: http://localhost/demo/opencats/index.php?m=home
[Wed Feb 11 13:34:59.884667 2015] [:error] [pid 1345] [client 127.0.0.1:55235] PHP Strict Standards:  Declaration of SimpleImageSubmitTag::write() should be compatible with SimpleWidget::write(&$encoding) in /var/www/html/demo/opencats/lib/simpletest/tag.php on line 483, referer: http://localhost/demo/opencats/index.php?m=home
[Wed Feb 11 13:34:59.885586 2015] [:error] [pid 1345] [client 127.0.0.1:55235] PHP Deprecated:  Assigning the return value of new by reference is deprecated in /var/www/html/demo/opencats/lib/simpletest/form.php on line 176, referer: http://localhost/demo/opencats/index.php?m=home
[Wed Feb 11 13:34:59.885634 2015] [:error] [pid 1345] [client 127.0.0.1:55235] PHP Deprecated:  Assigning the return value of new by reference is deprecated in /var/www/html/demo/opencats/lib/simpletest/form.php on line 195, referer: http://localhost/demo/opencats/index.php?m=home
[Wed Feb 11 13:34:59.886449 2015] [:error] [pid 1345] [client 127.0.0.1:55235] PHP Deprecated:  Assigning the return value of new by reference is deprecated in /var/www/html/demo/opencats/lib/simpletest/user_agent.php on line 48, referer: http://localhost/demo/opencats/index.php?m=home
```

**Comment by RussH** [comment]
I'll run a test-install myself today, and will fork - as the errors below
relate to inclusion of an ampersand, which is depregated in PHP5.x;

​
PHP Deprecated: Assigning the return value of new by reference is deprecated

**Comment by RussH** [comment]
All fixed by latest commit in 0.9.2 - updated the simpletest library to v1.1

**Comment by skrchnavy** [comment]
@RussH I understood this issue can be closed as fixed.
